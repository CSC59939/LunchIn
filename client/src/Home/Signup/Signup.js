import React, { Component } from "react";
import "./Signup.css";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  Divider,
  IconButton,
  InputAdornment,
  Snackbar,
  TextField,
  Typography,
  Tooltip
} from "@material-ui/core";
import CssBaseline from "@material-ui/core/CssBaseline";
import Visible from "@material-ui/icons/Visibility";
import Hidden from "@material-ui/icons/VisibilityOff";
import Check from "@material-ui/icons/CheckCircle";
import { firebase } from "../../Config";
import Geocode from "react-geocode";

const validation = {
  invalid: 0,
  valid: 1
};

export default class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userType: "user",
      name: "",
      email: "",
      password: "",
      address: "",
      lat: 0,
      lng: 0,
      showPassword: false,
      step1complete: false,
      step2complete: false,
      processing: false,
      notify: false,
      notifyMsg: "",
      message: "Show Password"
    };
    this.handleChange = this.handleChange.bind(this);
    this.createUser = this.createUser.bind(this);
    this.validate = this.validate.bind(this);
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  validate(type) {
    if (type === "createUser") {
      if (
        this.state.name === "" ||
        this.state.email === "" ||
        this.state.password === ""
      ) {
        return validation.invalid;
      }
    }
    return validation.valid;
  }
  createUser() {
    const isValid = this.validate("createUser");
    if (isValid === validation.valid) {
      this.setState({
        processing: true
      });

      Geocode.setApiKey(process.env.REACT_APP_GOOGLE_MAPS_API_KEY);
      console.log(this.state.address);
      Geocode.fromAddress(this.state.address).then(
        response => {
          const { lat, lng } = response.results[0].geometry.location;
          console.log(lat, lng);

          var setData = {
            type: this.state.userType,
            name: this.state.name,
            address: this.state.address,
            lat: lat,
            lng: lng
          };
          firebase
            .auth()
            .createUserWithEmailAndPassword(this.state.email, this.state.password)
            .then(() => {
              firebase
                .database()
                .ref(`Users/${firebase.auth().currentUser.uid}/`)
                .set(setData)
                .then(() => {
                  this.setState({
                    processing: false,
                    step1complete: true
                  }).then(() => {
                    this.props.history.push("/user/home");
                  });
                })
                .catch(error => {
                  this.setState({
                    notify: true,
                    notifyMsg: error.message,
                    processing: false,
                    step1complete: false
                  });
                });
            })
            .catch(error => {
              this.setState({
                notify: true,
                notifyMsg: error.message,
                processing: false,
                step1complete: false
              });
            });

        },
        error => {
          if (error.message === "Server returned status code ZERO_RESULTS") {
            console.error(error.message)
            this.setState({
              processing: false,
              notify: true,
              notifyMsg: "Invalid address. Please enter a valid address"
            });
          }
          if (error.message === "Server returned status code OVER_QUERY_LIMIT") {
            console.error(error.message)
            this.setState({
              processing: false,
              notify: true,
              notifyMsg: "Please try agian later"
            });
          }
        })

      // var setData = {
      //   type: this.state.userType,
      //   name: this.state.name,
      //   address: this.state.address
      // };
      // firebase
      //   .auth()
      //   .createUserWithEmailAndPassword(this.state.email, this.state.password)
      //   .then(() => {
      //     firebase
      //       .database()
      //       .ref(`Users/${firebase.auth().currentUser.uid}/`)
      //       .set(setData)
      //       .then(() => {
      //         this.setState({
      //           processing: false,
      //           step1complete: true
      //         }).then(() => {
      //           this.props.history.push("/user/home");
      //         });
      //       })
      //       .catch(error => {
      //         this.setState({
      //           notify: true,
      //           notifyMsg: error.message,
      //           processing: false,
      //           step1complete: false
      //         });
      //       });
      //   })
      //   .catch(error => {
      //     this.setState({
      //       notify: true,
      //       notifyMsg: error.message,
      //       processing: false,
      //       step1complete: false
      //     });
      //   });
    } else {
      this.setState({
        notify: true,
        notifyMsg: "Looks like you're missing stuff."
      });
    }
  }
  render() {
    return (
      <div className="signup-page">
        <CssBaseline />
        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={6000}
          message={this.state.notifyMsg}
        />
        <div className="signup-title-bar">
          <Typography variant="display2">LunchIn | Sign-Up</Typography>
          <Typography variant="subheading">
            Fill out the form below to sign-up for an account.
          </Typography>
        </div>
        <Divider />
        <div>
          <Card
            className="signup-page-content"
            style={{ marginTop: "5px" }}
            data-aos="slide-up"
          >
            <CardHeader
              title="Essentials"
              subheader="What we need to get started"
            />
            <CardContent>
              <TextField
                onChange={this.handleChange("name")}
                id="name"
                label="Full Name"
                required
                value={this.state.name}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                onChange={this.handleChange("email")}
                label="E-Mail"
                type="email"
                id="email"
                required
                value={this.state.email}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                onChange={this.handleChange("address")}
                id="address"
                label="Full Address"
                required
                value={this.state.address}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                disabled={this.state.step1complete || this.state.processing}
                id="password"
                label="Password"
                value={this.state.password}
                onChange={this.handleChange("password")}
                fullWidth
                required
                type={this.state.showPassword ? "text" : "password"}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <Tooltip title={this.state.message} placement="top">
                        <IconButton
                          onClick={() => {
                            if (this.state.message === "Show Password") {
                              this.setState({
                                showPassword: !this.state.showPassword,
                                message: "Hide Password"
                              });
                            } else if (this.state.message === "Hide Password") {
                              this.setState({
                                showPassword: !this.state.showPassword,
                                message: "Show Password"
                              });
                            }
                          }}
                        >
                          {this.state.showPassword ? <Visible /> : <Hidden />}
                        </IconButton>
                      </Tooltip>
                    </InputAdornment>
                  )
                }}
                className="push-down"
              />
              <Button
                onClick={() => {
                  this.createUser();
                }}
                fullWidth
                variant="raised"
                color="primary"
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              >
                <Check style={{ marginRight: "10px" }} />
                Sign Up
              </Button>
              <Button
                onClick={() => {
                  this.props.history.push(`/`);
                }}
                fullWidth
                variant="outlined"
                color="default"
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              >
                Already have an account? Sign in here!
              </Button>
            </CardContent>
          </Card>
          {this.state.step1complete ? (
            <Card className="push-down signup-page-content" data-aos="slide-up">
              <CardHeader
                title="Confirmation"
                subheader="Click done to create your account!"
              />
            </Card>
          ) : null}
        </div>
      </div>
    );
  }
}
