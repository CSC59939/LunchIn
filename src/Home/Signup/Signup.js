import React, { Component } from "react";
import "./Signup.css";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  Divider,
  IconButton,
  InputAdornment,
  Snackbar,
  TextField,
  Typography
} from "@material-ui/core";
import Visible from "@material-ui/icons/Visibility";
import Hidden from "@material-ui/icons/VisibilityOff";
import Next from "@material-ui/icons/PlayCircleFilled";
import Done from "@material-ui/icons/CheckCircle";
import { Link } from "react-router-dom";
import * as firebase from "firebase";

const validation = {
  invalid: 0,
  valid: 1,
  err_pwd_length: 2,
}


export default class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userType: "user",
      name: "",
      email: "",
      password: "",
      address: "",
      showPassword: false,
      step1complete: false,
      step2complete: false,
      processing: false,
      notify: false,
      notifyMsg: ""
    };
    this.handleChange = this.handleChange.bind(this);
    this.createUser = this.createUser.bind(this);
    this.finalizeUserUpdate = this.finalizeUserUpdate.bind(this);
    this.validate = this.validate.bind(this);
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };
  validate(type) {
    if (type === "createUser") {
      if (!this.state.name || !this.state.email || !this.state.password) {
        return validation.invalid;
      }
      if (this.state.password.length < 6) {
        return validation.err_pwd_length;
      }
    }
    return validation.valid;
  }
  createUser() {
    const isValid = this.validate("createUser");
    if (isValid === validation.valid) {
      this.setState({
        processing: true
      });
      firebase
        .auth()
        .createUserWithEmailAndPassword(this.state.email, this.state.password)
        .then(
          resp => {
            firebase.auth().currentUser.updateProfile({
              displayName: this.state.name
            });
          },
          this.setState({
            processing: false,
            step1complete: true
          }),
          err => {
            this.setState({
              notify: true,
              notifyMsg: err.message,
              processing: false
            });
          }
        );
    } else if(isValid === validation.err_pwd_length) {
      this.setState({
        notify: true,
        notifyMsg: "Password must at least 6 characters."
      });
    }
    else {
      this.setState({
        notify: true,
        notifyMsg: "Looks like you're missing stuff."
      });
    }
  }
  finalizeUserUpdate() {
    const isValid = this.validate("finalizeUser");
    if (isValid=== validation.valid) {
      this.setState({
        processing: true
      });
      var setData = {
        type: this.state.userType,
        name: this.state.name,
        address: this.state.address
      };
      firebase
        .database()
        .ref(`Users/${firebase.auth().currentUser.uid}/`)
        .set(setData)
        .then(
          () => {
            this.setState({
              processing: false,
              step2complete: true
            });
          },
          err => {
            this.setState({
              notify: true,
              notifyMsg: err.message,
              processing: false
            });
            return;
          }
        );
    } else if(isValid === validation.err_pwd_length) {
      this.setState({
        notify: true,
        notifyMsg: "Password must at least 6 characters."
      });
    }
     else {
      this.setState({
        notify: true,
        notifyMsg: "Looks like you're missing stuff.",
        processing: false
      });
    }
  }
  render() {
    return (
      <div className="signup-page">
        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={6000}
          message={this.state.notifyMsg}
        />
        <div className="signup-title-bar">
          <Typography variant="display2">LunchIn | Sign-Up</Typography>
          <Typography variant="subheading" style={{ marginTop: "10px" }}>
            Fill out the form below to sign-up for an account.
          </Typography>
        </div>
        <Divider />
        <div className="signup-page-content">
          <Card style={{ marginTop: "10px" }} data-aos="slide-up">
            <CardHeader
              title="Essentials"
              subheader="What we need to get started"
            />
            <CardContent>
              <TextField
                onChange={this.handleChange("name")}
                id="name"
                label="Full Name"
                required
                value={this.state.name}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                onChange={this.handleChange("address")}
                id="address"
                label="Full Address"
                required
                value={this.state.address}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                onChange={this.handleChange("email")}
                label="E-Mail"
                type="email"
                id="email"
                required
                value={this.state.email}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                disabled={this.state.step1complete || this.state.processing}
                id="password"
                label="Password"
                value={this.state.password}
                onChange={this.handleChange("password")}
                fullWidth
                required
                type={this.state.showPassword ? "text" : "password"}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <IconButton
                        onClick={() => {
                          this.setState({
                            showPassword: !this.state.showPassword
                          });
                        }}
                      >
                        {this.state.showPassword ? <Visible /> : <Hidden />}
                      </IconButton>
                    </InputAdornment>
                  )
                }}
                className="push-down"
              />
              <Button
                onClick={() => {
                  this.createUser();
                }}
                fullWidth
                variant="raised"
                color="primary"
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              >
                <Next style={{ marginRight: "10px" }} />
                Continue
              </Button>
            </CardContent>
          </Card>
          {this.state.step1complete ? (
            <Card className="push-down" data-aos="slide-up">
              <CardHeader
                title="Confirmation"
                subheader="Click done to create your account!"
              />
              <CardContent>
                <Button
                  onClick={() => {
                    this.finalizeUserUpdate();
                  }}
                  fullWidth
                  variant="raised"
                  color="primary"
                  className="push-down"
                  component={Link}
                  to="/"
                  disabled={this.state.step2complete || this.state.processing}
                >
                  <Done style={{ marginRight: "10px" }} />
                  Done
                </Button>
              </CardContent>
            </Card>
          ) : null}
        </div>
      </div>
    );
  }
}
