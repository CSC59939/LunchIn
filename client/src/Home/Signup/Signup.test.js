import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Signup from "./Signup";

describe("Signup Component", () => {
  it("Should render correctly with correct path", () => {
    const SignupInstance = shallow(<Signup />);
    expect(SignupInstance).toMatchSnapshot();
  });
  describe("When provided with user info", () => {
    it("Should render correctly", () => {
      const name = "Test";
      const email = "test@example.com"
      const address = "160 Convent Ave, New York, NY 10031"
      const password = "password"
      const lat = 40.8216144
      const lng = -73.9479427
      const SignupInstance = shallow(
        <Signup 
            name={name} 
            email={email}
            address={address}
            password={password}
            lat={lat}
            lng={lng}
        />
      );
      expect(SignupInstance).toMatchSnapshot();
    });
  });
  describe("When provided with empty address info", () => {
    it("Should render correctly", () => {
      const name = "Test";
      const email = "test@example.com"
      const password = "password"
      const address = " "
      const SignupInstance = shallow(
        <Signup  
            name={name} 
            email={email}
            address={address}
            password={password}
        />
      );
      expect(SignupInstance).toMatchSnapshot();
    });
  });
  describe("When provided with invalid address info", () => {
    it("Should render correctly", () => {
      const name = "Test";
      const email = "test@example.com"
      const password = "password"
      const address = "alskdfhwoei"
      const SignupInstance = shallow(
        <Signup  
            name={name} 
            email={email}
            address={address}
            password={password}
        />
      );
      expect(SignupInstance).toMatchSnapshot();
    });
  });
});
