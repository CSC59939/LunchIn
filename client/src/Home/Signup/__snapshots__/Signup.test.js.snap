// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component Should render correctly with correct path 1`] = `null`;

exports[`Signup Component When provided with empty address info Should render correctly 1`] = `
<div
  className="signup-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={6000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="signup-title-bar"
  >
    <WithStyles(Typography)
      component="h2"
      variant="h3"
    >
      LunchIn | Sign-Up
    </WithStyles(Typography)>
    <WithStyles(Typography)
      component="h2"
      variant="subtitle1"
    >
      Fill out the form below to sign-up for an account.
    </WithStyles(Typography)>
  </div>
  <WithStyles(Divider) />
  <div>
    <WithStyles(Card)
      className="signup-page-content"
      data-aos="slide-up"
      style={
        Object {
          "marginTop": "5px",
        }
      }
    >
      <WithStyles(CardHeader)
        subheader="What we need to get started"
        title="Essentials"
      />
      <WithStyles(CardContent)>
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="name"
          label="Full Name"
          onChange={[Function]}
          required={true}
          select={false}
          value=""
          variant="standard"
        />
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="email"
          label="E-Mail"
          onChange={[Function]}
          required={true}
          select={false}
          type="email"
          value=""
          variant="standard"
        />
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="address"
          label="Full Address"
          onChange={[Function]}
          required={true}
          select={false}
          value=""
          variant="standard"
        />
        <TextField
          InputProps={
            Object {
              "endAdornment": <WithStyles(InputAdornment)
                position="end"
              >
                <WithStyles(Tooltip)
                  placement="top"
                  title="Show Password"
                >
                  <WithStyles(IconButton)
                    onClick={[Function]}
                  >
                    <pure(VisibilityOffIcon) />
                  </WithStyles(IconButton)>
                </WithStyles(Tooltip)>
              </WithStyles(InputAdornment)>,
            }
          }
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="password"
          label="Password"
          onChange={[Function]}
          required={true}
          select={false}
          type="password"
          value=""
          variant="standard"
        />
        <WithStyles(Button)
          className="push-down"
          color="primary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          <pure(CheckCircleIcon)
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          Sign Up
        </WithStyles(Button)>
        <WithStyles(Button)
          className="push-down"
          color="default"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="outlined"
        >
          Already have an account? Sign in here!
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;

exports[`Signup Component When provided with invalid address info Should render correctly 1`] = `
<div
  className="signup-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={6000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="signup-title-bar"
  >
    <WithStyles(Typography)
      component="h2"
      variant="h3"
    >
      LunchIn | Sign-Up
    </WithStyles(Typography)>
    <WithStyles(Typography)
      component="h2"
      variant="subtitle1"
    >
      Fill out the form below to sign-up for an account.
    </WithStyles(Typography)>
  </div>
  <WithStyles(Divider) />
  <div>
    <WithStyles(Card)
      className="signup-page-content"
      data-aos="slide-up"
      style={
        Object {
          "marginTop": "5px",
        }
      }
    >
      <WithStyles(CardHeader)
        subheader="What we need to get started"
        title="Essentials"
      />
      <WithStyles(CardContent)>
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="name"
          label="Full Name"
          onChange={[Function]}
          required={true}
          select={false}
          value=""
          variant="standard"
        />
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="email"
          label="E-Mail"
          onChange={[Function]}
          required={true}
          select={false}
          type="email"
          value=""
          variant="standard"
        />
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="address"
          label="Full Address"
          onChange={[Function]}
          required={true}
          select={false}
          value=""
          variant="standard"
        />
        <TextField
          InputProps={
            Object {
              "endAdornment": <WithStyles(InputAdornment)
                position="end"
              >
                <WithStyles(Tooltip)
                  placement="top"
                  title="Show Password"
                >
                  <WithStyles(IconButton)
                    onClick={[Function]}
                  >
                    <pure(VisibilityOffIcon) />
                  </WithStyles(IconButton)>
                </WithStyles(Tooltip)>
              </WithStyles(InputAdornment)>,
            }
          }
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="password"
          label="Password"
          onChange={[Function]}
          required={true}
          select={false}
          type="password"
          value=""
          variant="standard"
        />
        <WithStyles(Button)
          className="push-down"
          color="primary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          <pure(CheckCircleIcon)
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          Sign Up
        </WithStyles(Button)>
        <WithStyles(Button)
          className="push-down"
          color="default"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="outlined"
        >
          Already have an account? Sign in here!
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;

exports[`Signup Component When provided with missing user info Should render correctly 1`] = `
<div
  className="signup-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={6000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="signup-title-bar"
  >
    <WithStyles(Typography)
      component="h2"
      variant="h3"
    >
      LunchIn | Sign-Up
    </WithStyles(Typography)>
    <WithStyles(Typography)
      component="h2"
      variant="subtitle1"
    >
      Fill out the form below to sign-up for an account.
    </WithStyles(Typography)>
  </div>
  <WithStyles(Divider) />
  <div>
    <WithStyles(Card)
      className="signup-page-content"
      data-aos="slide-up"
      style={
        Object {
          "marginTop": "5px",
        }
      }
    >
      <WithStyles(CardHeader)
        subheader="What we need to get started"
        title="Essentials"
      />
      <WithStyles(CardContent)>
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="name"
          label="Full Name"
          onChange={[Function]}
          required={true}
          select={false}
          value=""
          variant="standard"
        />
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="email"
          label="E-Mail"
          onChange={[Function]}
          required={true}
          select={false}
          type="email"
          value=""
          variant="standard"
        />
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="address"
          label="Full Address"
          onChange={[Function]}
          required={true}
          select={false}
          value=""
          variant="standard"
        />
        <TextField
          InputProps={
            Object {
              "endAdornment": <WithStyles(InputAdornment)
                position="end"
              >
                <WithStyles(Tooltip)
                  placement="top"
                  title="Show Password"
                >
                  <WithStyles(IconButton)
                    onClick={[Function]}
                  >
                    <pure(VisibilityOffIcon) />
                  </WithStyles(IconButton)>
                </WithStyles(Tooltip)>
              </WithStyles(InputAdornment)>,
            }
          }
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="password"
          label="Password"
          onChange={[Function]}
          required={true}
          select={false}
          type="password"
          value=""
          variant="standard"
        />
        <WithStyles(Button)
          className="push-down"
          color="primary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          <pure(CheckCircleIcon)
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          Sign Up
        </WithStyles(Button)>
        <WithStyles(Button)
          className="push-down"
          color="default"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="outlined"
        >
          Already have an account? Sign in here!
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;

exports[`Signup Component When provided with user info Should render correctly 1`] = `
<div
  className="signup-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={6000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="signup-title-bar"
  >
    <WithStyles(Typography)
      component="h2"
      variant="h3"
    >
      LunchIn | Sign-Up
    </WithStyles(Typography)>
    <WithStyles(Typography)
      component="h2"
      variant="subtitle1"
    >
      Fill out the form below to sign-up for an account.
    </WithStyles(Typography)>
  </div>
  <WithStyles(Divider) />
  <div>
    <WithStyles(Card)
      className="signup-page-content"
      data-aos="slide-up"
      style={
        Object {
          "marginTop": "5px",
        }
      }
    >
      <WithStyles(CardHeader)
        subheader="What we need to get started"
        title="Essentials"
      />
      <WithStyles(CardContent)>
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="name"
          label="Full Name"
          onChange={[Function]}
          required={true}
          select={false}
          value=""
          variant="standard"
        />
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="email"
          label="E-Mail"
          onChange={[Function]}
          required={true}
          select={false}
          type="email"
          value=""
          variant="standard"
        />
        <TextField
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="address"
          label="Full Address"
          onChange={[Function]}
          required={true}
          select={false}
          value=""
          variant="standard"
        />
        <TextField
          InputProps={
            Object {
              "endAdornment": <WithStyles(InputAdornment)
                position="end"
              >
                <WithStyles(Tooltip)
                  placement="top"
                  title="Show Password"
                >
                  <WithStyles(IconButton)
                    onClick={[Function]}
                  >
                    <pure(VisibilityOffIcon) />
                  </WithStyles(IconButton)>
                </WithStyles(Tooltip)>
              </WithStyles(InputAdornment)>,
            }
          }
          className="push-down"
          disabled={false}
          fullWidth={true}
          id="password"
          label="Password"
          onChange={[Function]}
          required={true}
          select={false}
          type="password"
          value=""
          variant="standard"
        />
        <WithStyles(Button)
          className="push-down"
          color="primary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          <pure(CheckCircleIcon)
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          Sign Up
        </WithStyles(Button)>
        <WithStyles(Button)
          className="push-down"
          color="default"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          variant="outlined"
        >
          Already have an account? Sign in here!
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;
